core/chains/evm/txmgr/eth_broadcaster.go

```diff
@@ -433,26 +430,14 @@ func (eb *EthBroadcaster) handleAnyInProgressEthTx(ctx context.Context, fromAddr
 	return nil, false
 }
 
-// getInProgressEthTx returns either 0 or 1 transaction that was left in
-// an unfinished state because something went screwy the last time. Most likely
-// the node crashed in the middle of the ProcessUnstartedEthTxs loop.
-// It may or may not have been broadcast to an eth node.
-func (eb *EthBroadcaster) getInProgressEthTx(fromAddress gethCommon.Address) (etx *EthTx, err error) {
-	etx = new(EthTx)
-	err = eb.q.Get(etx, `SELECT * FROM eth_txes WHERE from_address = $1 and state = 'in_progress'`, fromAddress.Bytes())
-	if errors.Is(err, sql.ErrNoRows) {
-		return nil, nil
-	} else if err != nil {
-		return nil, errors.Wrap(err, "getInProgressEthTx failed while loading eth tx")
-	}
-	if err = eb.orm.LoadEthTxAttempts(etx); err != nil {
-		return nil, errors.Wrap(err, "getInProgressEthTx failed while loading EthTxAttempts")
+// this function is used to pass the queryer from the txmgr to the keystore
+// it is inevitable we have to pass the queryer because we need none increment to happen in atomic transaction
+// but that happens in the keystore orm rather than the txmgr orm
+func (eb *EthBroadcaster) incrementNonceAtomic(tx pg.Queryer, etx EthTx) error {
```

Comment:
```suggestion
func (eb *EthBroadcaster) incrementNextNonceAtomic(tx pg.Queryer, etx EthTx) error {
```

Reply:
Why can't this be moved to the ORM to avoid the confusion of passing a callback?

Reply:
It comes down to 

```
func (eb *EthBroadcaster) incrementNextNonce(address gethCommon.Address, currentNonce int64, qopts ...pg.QOpt) error {
	return eb.ChainKeyStore.keystore.IncrementNextNonce(address, &eb.chainID, currentNonce, qopts...)
}
```

where you see that the keystore is responsible for incrementing the next nonce. Keystore has its own orm (method here https://github.com/smartcontractkit/chainlink/blob/c595b0e66b1956ab11f652bbc9bcbc33ce89c868/core/services/keystore/orm.go). 

I don't think we should move the keystore orm methods into the txmgr orm, so the only option I can think of is to pass the function which accepts the tx. Open to any ideas though

Reply:
We can also just duplicate the SQL code in the ORM. I might do that if there's no objections

Reply:
Yeah callback is probably the only way to do it